File: Algorithms/student-pack-stuff/matlab/get_random_location.m
function [random_position] = get_random_location( rows, cols, map)
isNotValid = true;

while (isNotValid)
    min_rows = 1;
    max_rows = max(rows);
    
    min_cols = 1;
    max_cols = max(cols);
    
    
    random_rows = round((max_rows - min_rows).*rand(1000,1) + min_rows);
    random_cols = round((max_cols - min_cols).*rand(1000,1) + min_cols);

    random_row = random_rows(1);
    random_col = random_cols(1);
    
    if map(random_row, random_col) == 0
        isNotValid = false;
    end
end
random_position = [random_row, random_col];

end

File: Algorithms/student-pack-stuff/c/getRandomLocation.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <functions.h>

#define MAX_ROWS 19
#define MAX_COLS 15

struct Location {
    int row;
    int col;
};

struct Location getRandomLocation(int map[MAX_ROWS][MAX_COLS]) {
    struct Location randomLocation;
    int isNotValid = 1; // true
    while(isNotValid) {
        int random_row = rand() % MAX_ROWS; // get random row and column
        int random_col = rand() % MAX_COLS; 
        // check point on map to see if it's a path
        if(map[random_row][random_col] == 0) {
            randomLocation.row = random_row;
            randomLocation.col = random_col;
            isNotValid = 0; 
        }
    }
    return randomLocation;    
}